impl/src/main/java/org/jboss/weld/bootstrap/FastAnnotatedTypeLoader.java

```diff
@@ -128,12 +126,21 @@
         return null;
     }
 
-    private boolean isJandexSetFlagsAvailable() {
-        try {
-            Method setFlags = AccessController.doPrivileged(GetDeclaredMethodAction.of(ClassFileInfo.class, "setFlags", short.class));
-            return setFlags != null;
-        } catch (Exception e) {
-            return false;
+    // checking availability of ClassInfo.setFlags method is just workaround for JANDEX-37
+    private boolean initCheckTypeModifiers() {
+
+        Class<?> classInfoclass = Reflections.loadClass(CLASSINFO_CLASS_NAME, new ClassLoaderResourceLoader(classFileServices.getClass().getClassLoader()));
```

Comment:
I am still not really sure how this will work in Servlet, SE environment but I don't want to think about it during weekend.:-) Will check at Monday.